.github/workflows/trigger_azdo_ci.yml

```diff
@@ -46,3 +46,11 @@ jobs:
         azure-devops-project-url: 'https://dev.azure.com/ni/DevCentral'
         azure-pipeline-name: 'ni-central-grpc_device-release-linux-rt'
         azure-devops-token: ${{ secrets.AZDO_PIPELINE_TRIGGERS }}
+
+    - name: Trigger CI
```

Comment:
I don't understand what's going on with the pipeline names and step names in this file. Everything is called "CI" and maybe has a branch name or an OS. Then this one is just "CI". And there's also a `ci.yml` in github.

CONSIDER if there's anything you can do with the PR descriptions, step names, or pipeline names to make this clearer. I know you're doing what you need to do to get the internal builds working; and I realize this isn't a new/easy problem.

Reply:
This new step will eventually replace the current steps shown on this yml file. This change is part of a [feature to improve grpc-device releases](https://ni.visualstudio.com/DevCentral/_workitems/edit/2507439). 

This isn't related to the ci.yml in github.

Reply:
`ci.yml` runs every time a change is pushed to `main` or `releases/*` or a PR targeting any branch is opened/synchronized/reopened.

`trigger_azdo_ci.yml` runs every time `ci.yml` completes. You could conceivably move it to the end of `ci.yml` instead of a separate workflow. I don't know if this is good or bad. If triggering AzDO fails, then `ci.yml` would fail.

Since `ci.yml` runs for PRs, `trigger_azdo_ci.yml` filters by branch name. However, the new "Trigger CI" job does not do this, so it triggers AzDO for PR builds too. @christag-ni Is that what you want?

In measurementlink-python and nidaqmx-python, I moved the on-PR trigger to a `PR.yml` which uses workflow_call to invoke `CI.yml`. The idea was that they are the same for now, but in the future we may want to customize them. Triggering AzDO is a good example of something we may want to do for on-push-to-main CI but not for PR builds.

Reply:
Ahh I see. This new "Trigger CI" job *should* run when any of the other `trigger_azdo_ci.yml` jobs normally run. Should I then change the filter to `if: ${{ github.repository == 'ni/grpc-device' && {github.event.workflow_run.head_branch == 'main' || startsWith(github.event.workflow_run.head_branch, 'releases')} }} `

?

Reply:
Yes, that would make sense.

Inside an expression, use parens instead of braces for logical grouping: https://docs.github.com/en/actions/learn-github-actions/expressions#operators

Reply:
Oops, I didn't notice this specifies [`on.workflow_run.branches`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore). So I think it's already filtered to only cover main and releases/*.